### DISCUSSION TITLE
CMake Error in CMakeLists.txt

### DISCUSSION THREAD

**Discussion by kitamuraa24** [Open-Sn/opensn#220] [Ask for Help] [answered]
Hello, 
I am trying to build and compile openSN, but when I try to run cmake i get the following error:

CMake Error in CMakeLists.txt:
Target "libopensn" INTERFACE_INCLUDE_DIRECTORIES property contains path:
"$USERPATH/opensn/dependencies/include"
which is prefixed in the source directory.

I added the environment path variables to my bashrc, if this is any help in understanding the problem. I am also building and compiling in a python venv.

export CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH:$SCRATCH/opensn/dependencies
export PETSC_DIR=$PETSC_DIR:$SCRATCH/opensn/dependencies
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$SCRATCH/dependencies/lib

Any advice is appreciated thanks.

**Comment by andrsd** [comment]
> export PETSC_DIR=$PETSC_DIR:$SCRATCH/opensn/dependencies

This is wrong. `PETSC_DIR` must be a single directory. If you installed PETSc into `$SCRATCH/opensn/dependencies`, then this should be:
```
export PETSC_DIR=$SCRATCH/opensn/dependencies
```

IDK, if that is gonna fix the problem you are seeing, thou.

As for `venv`: when you activate the environment, make sure that the variables like `CMAKE_PREFIX_PATH`, `PETSC_DIR`, etc. are set and point to the expected locations. This can be done by running `env` from a terminal.

Also, if you happen to switch environments between running cmake, it may confuse cmake (it likes to cache variable values). Then, you can remove all the files in the build dir and start from fresh. Note, that cmake complains about `$USERPATH`, so double check that variable in your environment and is set to expected value.

**Reply by kitamuraa24** [reply]
Hmm I checked all the boxes and it appears to be a different issue. Thank you though!

**Reply by andrsd** [reply]
Ok. Thanks for letting us known.

**Comment by kitamuraa24** [comment]
Ok I figured out the problem to the original issue. I needed to move the location of my dependencies outside of the main directory. It appears that I cannot have the dependencies in the same path as my source. However, I have a new issue when trying to compile.![image](https://github.com/Open-Sn/opensn/assets/157394294/76a1d87f-ee52-4628-a55e-a1a4cf13b82f)

**Reply by andrsd** [reply]
What compiler are you using? `cmake` reports that quite early during the "configure" step. 
The fix for this is trivial, but I would like to try to reproduce this problem...

**Reply by kitamuraa24** [reply]
I am building this on an HPC system, so I am module loading GCC/13.2.0.

**Reply by ragusa** [reply]
We state in the install notes not to build the dependencies in the same directory. Maybe we should state it with stronger language.

**Reply by kitamuraa24** [reply]
Ahh yea I was using the quick install guide, where it didn't specify that. My bad

**Reply by ragusa** [reply]
Any feedback on how we can improve the instructions is welcome. Please make note of what you think could be make better and collect all info into a feedback reply once you are done. Thanks

**Reply by andrsd** [reply]
> I am building this on an HPC system, so I am module loading GCC/13.2.0.

Out CI system currently uses gcc 9. I tested with gcc 11 on my linux machine. Both should give you a successful build. So, if your HPC system offers either version, maybe try that in the meantime until we patch the code for gcc 13.

**Reply by ragusa** [reply]
We have talked about CI with more builds. Until we can do this, should we be more explicit on the versions we use to build the code?

**Reply by wdhawkins** [reply]
I'm working on adding additional build configs now. I should have pretty decent coverage in the CI later today or early tomorrow.

**Reply by kitamuraa24** [reply, marked as answer]
Ok I successfully built/compiled using an older GCC. Thank you for the help!

**Reply by wdhawkins** [reply]
I've verified the error with GCC 13.x and submitted a fix. #223
